tracing-tracy/src/lib.rs

```diff
@@ -154,7 +177,8 @@ where
 
         let mut extensions = span.extensions_mut();
         if extensions.get_mut::<FormattedFields<F>>().is_none() {
-            let mut fields = FormattedFields::<F>::new(String::with_capacity(64));
+            let mut fields =
+                FormattedFields::<F>::new(CACHE.with(|cache| cache.pop().unwrap_or_default()));
```

Comment:
Consider adding a newtype that pushes contained string back into the cache on drop. As written if `format_fields` is not `Ok`, the buffer will be returned to heap.

Reply:
I also feel like it would perhaps make sense to keep `with_capacity` to reduce reallocations for the very first strings used here. Admittedly `64` was a guesstimate, but it does save on potentially, like, 5 reallocs?

Reply:
Whoops, great catch thanks! I added a guard type and brought back the 64 byte allocation as that does save a few hundred reallocs in my testing.